jgt_common/check.py

```diff
@@ -0,0 +1,184 @@
+"""
+Convenience functions for doing checks with helpful names and helpful messages.
+
+All the functions in this module return the empty string("") when the check passes
+and a description of how the check has failed when it fails.
+This is intentional so that the truthyness of strings and falsyness of empty strings
+can be exploited for use with ``format_if`` or ``assert`` or ...
+"""
+
+# NOTE TO IMPLEMENTORS:
+# While this module is stand-alone, it is also intended to work with
+# sibling module ``assert_``.
+# As such a number of conventions are expected of the code here:
+# 1 Keep to the promise made in the doc string above about strings and truthyness.
+# 2 Use the word "Check" at the start of the doc string so that the ``assert_`` module
+#   can make a sensible substituion.
+# 3 Any function defined in this module without a leading underscore honors #1.
+# 4 Corrollary to #3, any function starting with a leading underscore is internal to
+#   this module.
+# 5 If any classes are defined in this module, they will not be processed by ``assert_``
+#   and may or may not need a hand written version in that module.
+# 6 If we ever switch to using __all__ for controlling what functions we export,
+#   the ``assert_`` module  will need to be changed.
+# 7 There are no direct self-tests for this module because everything here
+#   is tested via the ``assert_`` module's tests.
+#   New functions added here should have new assertion tests in tests/test_asserts.py
+
+from jgt_common import percent_diff
+from . import format_if as _format_if
```

Comment:
Any reason one of these is `from jgt_common` and one is `from .`? Non-blocking; just curious.

Reply:
hmmm not sure as it was copied over from assert_.py so I'd like to clean it up as part of this PR.